### DISCUSSION TITLE
Unique fields in model?

### DISCUSSION THREAD

**Discussion by Atlinx** [twinlogix/typetta#212] [Q&A] [answered]
Is it possible to make a group of fields unique in a model, kinda like how Prisma has the [`@unique` and `@@unique`](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#unique-1) decorators? So far I'm thinking of using middleware to achieve this behaviour.

**Comment by edobrb** [comment, marked as answer]
This feature is not available yet. How would you achieve this with a middleware? The only way I can think of is to check in the before insert  if the record is already present, and to disable the update of the aforesaid fields, perhaps all fo this in transaction.

I think the best way to achieve this is to create the related indexes for MongoDB and related constraint for SQL.

**Reply by minox86** [reply]
I think the best way to achieve this is by a database constraint, but it depends on the behaviour you would like to have when the constraint is violated.

**Reply by Atlinx** [reply]
> This feature is not available yet. How would you achieve this with a middleware? The only way I can think of is to check in the before insert if the record is already present, and to disable the update of the aforesaid fields, perhaps all fo this in transaction.

Yep, that's exactly what I intended to do. I added a `@schema` decorator to my `typedefs` which lets me mark certain fields as unique.
